components/connectivity-adapter/internal/appregistry/service/graphql.go

```diff
@@ -28,44 +36,85 @@ type gqlGetApplicationResponse struct {
 	Result *graphql.ApplicationExt `json:"result"`
 }
 
-type gqlRequestBuilder struct {
+type gqlRequester struct {
+	cli               gqlcli.GraphQLClient
 	graphqlizer       GraphQLizer
 	gqlFieldsProvider GqlFieldsProvider
 }
 
-func NewGqlRequestBuilder(graphqlizer GraphQLizer, gqlFieldsProvider GqlFieldsProvider) *gqlRequestBuilder {
-	return &gqlRequestBuilder{graphqlizer: graphqlizer, gqlFieldsProvider: gqlFieldsProvider}
+func NewGqlRequester(cli gqlcli.GraphQLClient, graphqlizer GraphQLizer, gqlFieldsProvider GqlFieldsProvider) *gqlRequester {
```

Comment:
What do u thing about name: `NewDirectorClient`?

Reply:
Instead of `cli`, we can pass `cliProvider`, then in query we can retrieve `cli`.
One downside, is that we have to pass `http.Request` to the method.

Reply:
Sounds good - it's fine for me. But please think about a better name for ServiceManager - that's way more important than this 😛